src/main/common/sensor_alignment.c

```diff
@@ -0,0 +1,95 @@
+/*
+ * This file is part of Cleanflight and Betaflight.
+ *
+ * Cleanflight and Betaflight are free software. You can redistribute
+ * this software and/or modify this software under the terms of the
+ * GNU General Public License as published by the Free Software
+ * Foundation, either version 3 of the License, or (at your option)
+ * any later version.
+ *
+ * Cleanflight and Betaflight are distributed in the hope that they
+ * will be useful, but WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+ * See the GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this software.
+ *
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#include "stdbool.h"
+
+#include "platform.h"
+
+#include "common/sensor_alignment.h"
+
+
+void buildRotationMatrixFromAlignment(const sensorAlignment_t* sensorAlignment, fp_rotationMatrix_t* rm)
+{
+    fp_angles_t rotationAngles;
+    rotationAngles.angles.roll  = DEGREES_TO_RADIANS(sensorAlignment->values.roll);
+    rotationAngles.angles.pitch = DEGREES_TO_RADIANS(sensorAlignment->values.pitch);
+    rotationAngles.angles.yaw   = DEGREES_TO_RADIANS(sensorAlignment->values.yaw);
+
+    buildRotationMatrix(&rotationAngles, rm);
+}
+
+bool isSensorAlignmentStandard(sensorAlignment_t* sensorAlignment)
+{
+    bool nonStandardAlignmentDetected = false;
+
+    for (int i = 0; i < XYZ_AXIS_COUNT && !nonStandardAlignmentDetected; i++) {
+        nonStandardAlignmentDetected = (int)sensorAlignment->raw[i] % 90 != 0;
+    }
+
+    return !nonStandardAlignmentDetected;
+}
+
+void updateStandardAlignmentFromNonDefaultLegacyRotation(sensorAlignment_t* sensorAlignment, sensor_align_e rotation)
```

Comment:
I disagree with the naming here - I don't think that non-orthogonal mounting of sensors will be more than a niche, and for orthogonal mounting using permutation is still the preferred and most efficient way, and by no means 'legacy'.
Also, in a broader sense, I do not know why we are building a `sensorAlignment_t` here from an orthogonal alignment, as we will want to use permutation to rotate in these cases, and not rotation matrix. If anything, we should turn this around and check if a custom alignment is orthogonal, and use permutation in this case for efficiency.

Reply:
yeah, I dropped the 'legacy' in most of the other code in this PR, this one was missed, but likely this code will be deleted as it's only used by test code and commented out MSP code so don't get hung-up on this point.

Reply:
@ledvinap yes the performance was checked in the previous PR and my @fiam in the related iNAV work.  I fully agree it's cleaner to have one and only one way of doing it, the main issue with not always using a rotation matrix is one of config.

Recently however we discussed updating the rotation matrix config settings automatically when the enum config setting is changed and vice versa, I will attempt that later in this PR.

I'm happy to have this dual-code-path solution for now as it allows progression towards a time when we only have one path and solves the upgrade/config 2-stage release issue.

Reply:
naming updated, 'standard' and 'custom' are used now.

the 'updateStandardAlignment...' here was used by the previous MSP code, which has now been removed.  This method, and some related ones have been moved to test code so that they can be used in `validateAndFixConfig` as per conversations with @mikeller  and @blckmn via slack.  They may yet be deleted.